src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/NonAssociation.kt

```diff
@@ -0,0 +1,122 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.integration.latestBooking
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.NonAssociationReason
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.NonAssociationType
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Offender
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderBooking
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderNonAssociation
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderNonAssociationDetail
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderNonAssociationId
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.ReferenceCode
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderNonAssociationRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.ReferenceCodeRepository
+import java.time.LocalDate
+
+@DslMarker
+annotation class NonAssociationDslMarker
+
+@NomisDataDslMarker
+interface NonAssociationDsl {
+  @NonAssociationDetailDslMarker
+  fun nonAssociationDetail(
+    typeSeq: Int = 1,
+    nonAssociationReason: String,
+    recipNonAssociationReason: String? = null,
+    nonAssociationType: String,
+    effectiveDate: LocalDate,
+    expiryDate: LocalDate? = null,
+    authorisedBy: String? = null,
+    comment: String? = null,
+  ): OffenderNonAssociationDetail
+}
+
+@Component
+class NonAssociationBuilderRepository(
+  private val nonAssociationReasonRepository: ReferenceCodeRepository<NonAssociationReason>? = null,
+  private val nonAssociationTypeRepository: ReferenceCodeRepository<NonAssociationType>? = null,
+  private val nonAssociationRepository: OffenderNonAssociationRepository? = null,
+) {
```

Comment:
I'm not sure why these are all nullable? That's a pattern used in places for unit tests but not here so they shouldn't be nullable.